src/main/scala/com/fulcrumgenomics/sv/tools/SvPileup.scala

```diff
@@ -297,9 +305,11 @@ object SvPileup extends LazyLogging {
    */
   def filterTemplate(t: Template,
                      minPrimaryMapq: Int,
-                     minSupplementaryMapq: Int): Option[Template] = {
-    val r1PrimaryOk = t.r1.exists(r => r.mapped && r.mapq >= minPrimaryMapq)
-    val r2PrimaryOk = t.r2.exists(r => r.mapped && r.mapq >= minPrimaryMapq)
+                     minSupplementaryMapq: Int,
+                     incDupes: Boolean,
+                     incQcFails: Boolean): Option[Template] = {
+    val r1PrimaryOk = t.r1.exists(r => r.mapped && r.mapq >= minPrimaryMapq && (incDupes || !r.duplicate) && (incQcFails || r.pf))
+    val r2PrimaryOk = t.r2.exists(r => r.mapped && r.mapq >= minPrimaryMapq && (incDupes || !r.duplicate) && (incQcFails || r.pf))
 
     if (!r1PrimaryOk && !r2PrimaryOk) None else Some(
       Template(
```

Comment:
It's surprising to me that one read but not both may be filtered. I would expect the filtering would be applied to templates and not individual reads - should this be updated to filter the entire template if one read fails the mapq check? (Or to keep the entire template - perhaps the desired behavior could be a CLI flag?)

Reply:
I think that's probably a topic for a follow-up PR if we prefer.  That said, I think it does make sense that _reads_ can be filtered independently based on mapq at least.

E.g. I think you'd want to retain reads in the following situations:

1. Read 1 is mapped at high mapq and read 2 is unmapped
2. Read 1 is mapped at high mapq and read 2 has low mapq because it's wholly contained within a simple (dinuc/trinuc) repeat

In both cases, if R1 has a split read alignment then that information is still useful, no?